app/data_models/relationship_store.py

```diff
@@ -21,45 +21,47 @@ class RelationshipStore:
     Stores and updates relationships.
     """
 
-    def __init__(self, relationships: Optional[List[Mapping]] = None) -> None:
+    def __init__(self, relationships: dict = None) -> None:
         self._is_dirty = False
         self._relationships = self._build_map(relationships or [])
 
-    def __iter__(self):
+    def __iter__(self) -> Iterator:
         return iter(self._relationships.values())
 
-    def __contains__(self, relationship):
+    def __contains__(self, relationship: Relationship) -> bool:
         return (
             relationship.list_item_id,
             relationship.to_list_item_id,
         ) in self._relationships
 
-    def __len__(self):
+    def __len__(self) -> int:
         return len(self._relationships)
 
     @property
-    def is_dirty(self):
+    def is_dirty(self) -> bool:
         return self._is_dirty
 
-    def clear(self):
+    def clear(self) -> None:
         self._relationships.clear()
         self._is_dirty = True
 
-    def serialize(self):
+    def serialize(self) -> list:
         return [
             relationship.for_json() for relationship in self._relationships.values()
         ]
 
-    def get_relationship(self, list_item_id, to_list_item_id):
+    def get_relationship(
+        self, list_item_id: str, to_list_item_id: str
+    ) -> Union[Relationship, list]:
```

Comment:
This should return an optional Relationship.

Reply:
Returning optional Relationship now, thanks.